Controller/ProductController.cs

```diff
@@ -0,0 +1,349 @@
+                var quantity = 0;
+                while (string.IsNullOrWhiteSpace(quantityInput))
+                {
+                    quantityInput = Console.ReadLine();
+                    if(quantityInput == "back") {
+                        Console.Clear();
+                        return;
+                    }
+                    if (string.IsNullOrWhiteSpace(quantityInput))
+                    {
+                        Console.WriteLine("Quantity cannot be empty, Please try again.");
+                    }
+                    else if (int.TryParse(quantityInput, out int q) && q > 0)
+                    {
+                        quantity = q;
+                    }
+                    else
+                    {
+                        Console.WriteLine("Please enter numeric value only greater than zero.");
+                        quantityInput = "";
+                    }
+                }
+
+                var product = _productService.GetAllProducts().FirstOrDefault(p => p.Name == name);
+
+                if (product is null)
+                {
+                    product = new Product() { Id = Guid.NewGuid(), Name = name, Price = price, Quantity = quantity };
+                    _productService.AddProduct(product);
+                    Console.WriteLine("Product added successfully.");
+                }
+                else
+                {
+                    Console.WriteLine("Product already exists, cannot be added");
+                }
+                Console.WriteLine("\nPress any key to return to menu.");
+                Console.ReadKey();
+                Console.Clear();
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine("Something went wrong.");
+                CustomLogger.Logger.LogError(e);
+            }
+        }
+
+        public void ShowAllProductsCustomer()
+        {
+            try
+            {
+                var products = _productService.GetAllProducts().Where(prod => prod.Quantity > 0).ToList();
+                if (products.Count <= 0)
+                {
+                    Console.WriteLine("No products available.");
+                    Console.WriteLine("\nPress any key to return to menu.");
+                    Console.ReadKey();
+                    Console.Clear();
+                    return;
+                }
+                int i = 1;
+                foreach (var product in products)
+                {
+                    Console.Write($"Sr. No. {i++},  ");
+                    product.DisplayInfo();
+                }
+                Console.WriteLine("\nPress any key to return to menu.");
+                Console.ReadKey();
+                Console.Clear();
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine("Something went wrong.");
+                CustomLogger.Logger.LogError(e);
+            }
+        }
+
+        public void ShowAllProductsAdmin()
```

Comment:
This method has common code with method 'ShowAllProductsCustomer', can you not extract common code and avoid duplication?

Reply:
ShowAllProductsAdmin??